lib/chess/pieces/piece.rb

```diff
@@ -0,0 +1,116 @@
+require_relative '../chess'
+require_relative '../step'
+require 'colorize'
+require 'colorized_string'
+require 'byebug'
+
+module Chess
+  class Piece
+    attr_reader :symbol, :team, :current_step
+
+    Vector = Struct.new(:enabled, :deltas)
+
+    def initialize(position_algebraic, symbol, team, captured = false)
+      @current_step = Step.new(position_algebraic)
+      @symbol = symbol
+      @captured = captured
+      @team = team
+      @enemies_team = ([WHITE_TEAM, BLACK_TEAM] - [@team]).first
+    end
+
+    def move_to(target_position_algebraic, position_deltas_vectors, occuped_cells = nil)
```

Comment:
- A better name `target_position_algebraic` if `algebraic_target_position`.
- `target_position_algebraic` should be an instance of `Position`, and even better, it should be agnostic to the type of position that you need in this case (algebraic or coordinates), those two should be methods inside the `Position` class

- `position_deltas_vectors` I think it could be renamed to `position_vectors`.

- `occuped_cells` -> `occupied_cells`
- I'd expect these `occupied_cells` to be an array of `Board::Cell`